vectorstores/weaviate/weaviate.go

```diff
@@ -197,10 +201,19 @@ func (s Store) parseDocumentsByGraphQLResponse(res *models.GraphQLResponse) ([]s
 		if !ok {
 			return nil, ErrMissingTextKey
 		}
+		additional, ok := itemMap["_additional"].(map[string]any)
```

Comment:
what are the conditions/guarantees that this is present?

Reply:
if it isn't always present it seems like we should return the document without a populated score vs return an error earlier, thoughts?

Reply:
> what are the conditions/guarantees that this is present?
`certainty` in `_additional` is weaviate's built-in attributes:
https://weaviate.io/developers/weaviate/api/graphql/additional-properties

and we add this in https://github.com/tmc/langchaingo/blob/e98eb1703a8d377ef3efeaa68fb24b33ef34e726/vectorstores/weaviate/weaviate.go#L262

but I agress we can return the document without a populated score.